script.js

```diff
@@ -1,12 +1,210 @@
+let _localEpisodes = [];
+let _localShows = [];
+let _showDDLCtrlId = "selShow";
+let _epDDLCtrlId = "selName";
+function sortByName(arr) {
+  return arr.sort(function (a, b) {
+    var textA = a.name.toLowerCase();
+    var textB = b.name.toLowerCase();
+    return textA < textB ? -1 : textA > textB ? 1 : 0;
+  });
+}
+async function getAllShowAsync() {
+  if (_localShows.length == 0) {
+    _localShows = sortByName(getAllShows());
+  }
+  return true;
+}
+async function getAllEpisodesAsync(showId) {
+  if (!showId && _localShows.length == 0) {
+    console.error("Cannot get shows");
+  }
+  let currentShowId = showId ?? _localShows[0].id;
+  let localShowsEpisodes = {};
+  //use local storage to avoid api load
+  let localShowsEpisodesJsonStr = localStorage.getItem("localShowsEpisodes");
+  //localStorage has localShowsEpisodes Item and the item has currentshowId key
+  if (localShowsEpisodesJsonStr) {
+    localShowsEpisodes = JSON.parse(localShowsEpisodesJsonStr);
+    if (localShowsEpisodes[currentShowId]) {
+      //setup Episodes
+      _localEpisodes = localShowsEpisodes[currentShowId];
+      return true;
+    }
+  }
+  //if localStorage items have no key equals to currentShowId, call api
+  const response = await fetch(
+    "https://api.tvmaze.com/shows/" + currentShowId + "/episodes"
+  );
+  //save the result to localStorage
+  localShowsEpisodes[currentShowId] = await response.json();
+  localStorage.setItem(
+    "localShowsEpisodes",
+    JSON.stringify(localShowsEpisodes)
+  );
+  //setup Episodes
+  _localEpisodes = localShowsEpisodes[currentShowId];
+  return true;
+}
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+async function setup() {
+  createSearchResult();
+  createShowMenu();
+  createEPMenu();
+  createSearchBar();
+  let allshowsResult = await getAllShowAsync();
+  if (allshowsResult) {
+    let allEpisodesResult = await refreshEpisodesDDL();
+    refreshShowMenu();
+    if (allEpisodesResult) {
+      drawEpiCard();
+    }
+  }
+}
```

Comment:
this is good that you are using a setup function to create all your initial elements - your page loads well and generates all the expected episodes from the first show in the list, so well done. 
There seems to be a bit of  problem with the `createSearchResult()` function as it isn't showing the total list of episodes (as it is called when `_localEpisodes.length` is still 0, so the result is always Got 0 episode(s))